scipy/stats/_discrepancy.pyx

```diff
@@ -0,0 +1,382 @@
+# cython: language_level=3
+# cython: boundscheck=False
+# cython: wraparound=False
+# cython: cdivision=True
+
+cimport cython
+import numpy as np
+cimport numpy as np
+from libc.math cimport fabs, sqrt, pow
+
+np.import_array()
+
+
+def discrepancy(sample, bint iterative=False, method='CD'):
```

Comment:
As this is just a python function, I would leave it in `_qmc` like we do with `Sobol`

Reply:
This function call `cdef` functions. It can not be directly in python (or some changes must be done).

Reply:
Yes the functions would have to be `cpdef`. But maybe the overhead would be ok.